src/orchestratord/src/controller/materialize/cockroach.rs

```diff
@@ -0,0 +1,223 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::BTreeMap;
+
+use k8s_openapi::api::core::v1::Secret;
+use kube::Api;
+use rand::{distributions::Alphanumeric, thread_rng, Rng};
+use tracing::{debug, trace};
+use urlencoding::encode;
+
+use crate::k8s::{apply_resource, get_resource};
+use mz_cloud_resources::crd::materialize::v1alpha1::Materialize;
+use mz_ore::instrument;
+
+const SCHEMAS: &[&str] = &["consensus", "tsoracle"];
+
+#[derive(clap::Parser)]
+pub struct CockroachInfo {
+    #[clap(long)]
+    pub cockroach_endpoint: String,
+    #[clap(long)]
+    pub cockroach_port: u16,
+    #[clap(long)]
+    pub cockroach_ca_cert: String,
+    #[clap(long)]
+    pub cockroach_username: String,
+    #[clap(long)]
+    pub cockroach_password: String,
```

Comment:
oh I feel like we should call out that this is WIP and not how we'll wire up creds for real

Reply:
done